closed/autoconf/custom-hook.m4

```diff
@@ -52,47 +52,17 @@ AC_DEFUN_ONCE([CUSTOM_EARLY_HOOK],
   OPENJ9_CONFIGURE_NUMA
   OPENJ9_CONFIGURE_SNAPSHOTS
   OPENJ9_CONFIGURE_WARNINGS
-  OPENJ9_THIRD_PARTY_REQUIREMENTS
   OPENJ9_CHECK_NASM_VERSION
   OPENJCEPLUS_SETUP
 ])
 
 AC_DEFUN([OPENJ9_CONFIGURE_CMAKE],
 [
-  AC_ARG_WITH(cmake, [AS_HELP_STRING([--with-cmake], [enable building openJ9 with CMake])],
-    [
-      if test "x$with_cmake" = xyes -o "x$with_cmake" = x ; then
-        with_cmake=cmake
-      fi
-    ],
-    [
-      case "$OPENJ9_PLATFORM_CODE" in
-        ap64|oa64|or64|wa64|xa64|xl64|xr64|xz64)
-          with_cmake=cmake
-          ;;
-        *)
-          with_cmake=no
-          ;;
-      esac
-    ])
-  # at this point with_cmake should either be no, or the name of the cmake command
-  if test "x$with_cmake" = xno ; then
-    OPENJ9_ENABLE_CMAKE=false
-
-    # Currently, mixedrefs mode is only available with CMake enabled
-    if test "x$OMR_MIXED_REFERENCES_MODE" != xoff ; then
-      AC_MSG_ERROR([[--with-mixedrefs=[static|dynamic] requires --with-cmake]])
-    fi
+  if AS_EXECUTABLE_P(cmake) ; then
+    CMAKE=cmake
   else
```

Comment:
Please explain this use of `AS_EXECUTABLE_P()` why not just
```
  UTIL_REQUIRE_PROGS([CMAKE], cmake)
```
to allow `CMAKE` from the environment to have effect?

Reply:
I don't understand it so I left it as it was.

Reply:
I'm thinking the idea was that you could specify a different cmake executable, but we don't seem to support that or use that. I'll remove it.

Reply:
Actually, I think this should be simpler still; `CMAKE` was already probed in `basic_tools.m4`, we just need to insist if was found:
```
 AC_DEFUN([OPENJ9_CONFIGURE_CMAKE],
[
  if test "x$CMAKE" = x ; then
    AC_MSG_ERROR([CMAKE is required to build OpenJ9])
  fi
])
```

Reply:
Changed.